Funcky/Extensions/ParseExtensions/ParseExtensions.TypeName.cs

```diff
@@ -0,0 +1,9 @@
+#if REFLECTION_TYPE_NAME
+using System.Reflection.Metadata;
+using Funcky.Internal;
+
+namespace Funcky.Extensions;
+
+[OrNoneFromTryPattern(typeof(TypeName), nameof(TypeName.TryParse))]
```

Comment:
I feel `System.Reflection.Metadata.TypeName` is fairly niche, should we really add the parse methods?

Reply:
I think we just do it, why adding some additional criteria. If it is in there, we support it.